### PR TITLE
Scripts for feature "OnRCStatus notification"

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
ATF Test Scripts to check #[1930](https://github.com/smartdevicelink/sdl_core/issues/1930)
This PR is **ready** for review.

### Summary
Scripts for feature "OnRCStatus notification" 

### ATF version
[Develop branch](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by yang1070** [comment]
when I run
```
zhimin@SYNC3:~/github/sdl/atf_run$ ../cycleFolderRun.sh test_scripts/RC/OnRCStatus/ --sdl-core=/home/zhimin/github/sdl/sdlb/bin/ | tee my_atf.log
```
I got the follwing
```
==============================
Start 'test_scripts/RC/OnRCStatus/019_notification_by_app_disconnect.lua'
==============================
 WARNING: Parameter pathToSDLInterfaces is not specified, default APIs are used 
SDL build option EXTENDED_POLICY is unavailable.
Assume that SDL was built with EXTENDED_POLICY = PROPRIETARY
SDL build option REMOTE_CONTROL is unavailable.
Assume that SDL was built with REMOTE_CONTROL = ON
```
Do we still need REMOTE_CONTROL = ON? with the refactored code?

File: test_scripts/RC/OnRCStatus/015_notification_by_user_exit.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---------------------------------------------------------------------------------------------------
+-- User story: TBD
+-- Use case: TBD
+--
+-- Requirement summary:
+-- [SDL_RC] TBD
+--
+-- Description: SDL shall send OnRCStatus notifications to rc registered app
```
Comment (yang1070) [inline test_scripts/RC/OnRCStatus/015_notification_by_user_exit.lua]: Description is not consistent with the file name says

File: test_scripts/RC/OnRCStatus/001_notification_by_apps_registration.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---------------------------------------------------------------------------------------------------
+-- User story: TBD
+-- Use case: TBD
+--
+-- Requirement summary:
+-- [SDL_RC] TBD
+--
+-- Description: SDL shall send OnRCStatus notifications to all registered mobile applications and the HMI
```
Comment (yang1070) [inline test_scripts/RC/OnRCStatus/001_notification_by_apps_registration.lua]: I like the description in PR https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1909/
That have more information on condition/steps/input to the sdl core, and the expected out put

**Comment by GetmanetsIrina** [comment]
@yang1070 , please find update regarding descriptions in b65d5dde15c335f9f74eab95c1f6b34f213d68f2

**Comment by GetmanetsIrina** [comment]
@yang1070 , according to `Do we still need REMOTE_CONTROL = ON? with the refactored code?` with the refactoring code we do not need REMOTE_CONTROL = ON, this functionality of ATF, not of scripts. 

For now this functionality is not affect scripts execution, fix for ATF will be provided when refactoring code will be merged.

**Comment by yang1070** [comment]
@GetmanetsIrina Thank you for the info regarding ATF and macro REMOTE_CONTROL =ON

File: test_scripts/RC/OnRCStatus/003_registration_not_rc_app.lua [hunk 3]
```diff
@@ -5,7 +5,11 @@
 -- Requirement summary:
 -- [SDL_RC] TBD
 --
--- Description: SDL shall not send OnRCStatus notifications to not rc registered app
+-- Description:
+-- In case:
+-- 1) Not RC application is registered
```
Comment (yang1070) [inline test_scripts/RC/OnRCStatus/003_registration_not_rc_app.lua]: Not RC application is registered ==> No RC application is registered

Comment (GetmanetsIrina) [reply test_scripts/RC/OnRCStatus/003_registration_not_rc_app.lua]: @yang1070 updated to non-Rc in 4f94bb15b488535907390542330636d3e45e5b58 , `no RC application is registered` is meaning not the same that i mean.

**Comment by GetmanetsIrina** [comment]
Was added 2 additional commits:
5c4b67dc1918ba1142dd937097f7b326217cfedb - fixed error script name in test set,
8c507f2663eb133e31775bb058f397cda5d45a29 - added generating permissions structure for application from default structure. This update is triggered by merge in develop the feature expandable_design_for_proprietary_data_exchange. 
PTU was failed because of absence RequestType and RequestSubType parameters in ptu from mobile.
Now permissions for app are created with RequestType and RequestSubType values from default section in case they are present in default.

**Pull request merged**